tensorly/decomposition/_parafac2.py

```diff
@@ -544,6 +602,18 @@ def parafac_updates(X, w, f):
                 rec_errors[-1],
             )
 
+        # Update imputations
+        if mask is not None:
+
+            reconstructed_tensor = parafac2_to_tensor((weights, factors, projections))
+
+            for idx in indices_missing:
+                tl.index_update(
+                    tensor_slices,
+                    tl.index[idx[0], idx[1], idx[2]],
```

Comment:
This will not work for order>3 tensors; there should be a better way do this?

In other functions we write something like tensor = tensor*mask + reconstructed_tensor * (1-mask)
which I think would also apply here.